### ISSUE TITLE
`--packages=external` option affects local files mapped by `tsconfig.json`'s `paths` option

### ISSUE THREAD

**Issue by reosablo** [evanw/esbuild#3238]
When using short hand imports like `import foo from "~/foo";` mapped by `tsconfig`'s `paths` option, it appears that they are treated as external dependencies when the `--packages=external` option is specified.

```jsonc
// tsconfig.json
{
  "compilerOptions": {
    "paths": {
      "~/*": ["./*"]
    }
  }
}
```

```js
// entry.js
import foo from "~/foo";
```

Current output: `~/foo` is not resolved as an external dependency. [demo](https://esbuild.github.io/try/#YgAwLjE4LjEzACAtLWJ1bmRsZSAtLWZvcm1hdD1lc20gLS1wbGF0Zm9ybT1ub2RlIC0tcGFja2FnZXM9ZXh0ZXJuYWwAZQBlbnRyeS5qcwBpbXBvcnQgZm9vIGZyb20gIn4vZm9vIjsKaW1wb3J0IHByb2Nlc3MgZnJvbSAibm9kZTpwcm9jZXNzIjsKCmZvbygpLmNhdGNoKCgpID0+IHByb2Nlc3MuZXhpdCgxKSk7AABmb28uanMAaW1wb3J0IGZ1bmMgZnJvbSAiZXh0IjsKCmV4cG9ydCBkZWZhdWx0IGFzeW5jIGZ1bmN0aW9uIGZvbygpIHsKICByZXR1cm4gYXdhaXQgZnVuYygpOwp9AAB0c2NvbmZpZy5qc29uAHsKICAiY29tcGlsZXJPcHRpb25zIjogewogICAgInBhdGhzIjogewogICAgICAifi8qIjogWyIuLyoiXQogICAgfQogIH0KfQ)

Expected output: `~/foo` is resolved. [demo (just replaced `--packages=external` with `--external:ext`)](https://esbuild.github.io/try/#YgAwLjE4LjEzACAtLWJ1bmRsZSAtLWZvcm1hdD1lc20gLS1wbGF0Zm9ybT1ub2RlIC0tZXh0ZXJuYWw6ZXh0AGUAZW50cnkuanMAaW1wb3J0IGZvbyBmcm9tICJ+L2ZvbyI7CmltcG9ydCBwcm9jZXNzIGZyb20gIm5vZGU6cHJvY2VzcyI7Cgpmb28oKS5jYXRjaCgoKSA9PiBwcm9jZXNzLmV4aXQoMSkpOwAAZm9vLmpzAGltcG9ydCBmdW5jIGZyb20gImV4dCI7CgpleHBvcnQgZGVmYXVsdCBhc3luYyBmdW5jdGlvbiBmb28oKSB7CiAgcmV0dXJuIGF3YWl0IGZ1bmMoKTsKfQAAdHNjb25maWcuanNvbgB7CiAgImNvbXBpbGVyT3B0aW9ucyI6IHsKICAgICJwYXRocyI6IHsKICAgICAgIn4vKiI6IFsiLi8qIl0KICAgIH0KICB9Cn0)

https://esbuild.github.io/api/#packages

**Comment by evanw** [comment]
Note to self: Someone has already reported this before, although they closed it: #3160

**Comment by evanw** [comment]
The other issues are closed but I need an open one to track this. Reopening this issue.
